### PR TITLE
Add error message for when following own account

### LINKED ISSUE #22690: Inaccurate error message when attempting to follow your own account via the API
### Steps to reproduce the problem

Attempt to follow your own account via the API

### Expected behaviour

An error should be returned like "Cannot follow your own account"

### Actual behaviour

404: Record not found

### Detailed description

I encountered this error while using a script to bulk import following accounts from an account that happened to follow me.

### Specifications

4.0.2

### PR DESCRIPTION
**Pull request by gabrielcosta42**
* Add a specific error message for when attempting to follow your own account via the API

It would handle the attempt to follow your own account as if the account record didn't exist.

* Add tests

fixes  #22690

### CODE REVIEW

**Comment by gunchleoc** [comment]
English language fix suggestions:

> Following your own account is not allowed

or

> You can’t follow your own account

**Comment by vmstan** [comment]
@gabrielcosta42 are you able to rebase this?

File: app/controllers/api/v1/accounts_controller.rb [hunk 1]
```diff
@@ -38,6 +38,9 @@ def follow
     options = @account.locked? || current_user.account.silenced? ? {} : { following_map: { @account.id => { reblogs: follow.show_reblogs?, notify: follow.notify?, languages: follow.languages } }, requested_map: { @account.id => false } }
 
     render json: @account, serializer: REST::RelationshipSerializer, relationships: relationships(**options)
+
+  rescue FollowService::SelfFollowError
+    render json: { error: 'Follow your own account is not allowed' }, status: 403
```
Comment (vmstan) [inline app/controllers/api/v1/accounts_controller.rb:43]:
```suggestion
    render json: { error: 'Following your own account is not allowed' }, status: 403
```

File: spec/controllers/api/v1/accounts_controller_spec.rb [hunk 2]
```diff
@@ -117,6 +117,27 @@
...
+        let(:other_account) { user.account }
+
+        it 'returns http forbidden' do
+          expect(response).to have_http_status(403)
+        end
+
+        it 'returns JSON with an error message' do
+          json = body_as_json
+          
+          expect(json[:error]).to eq 'Follow your own account is not allowed'
```
Comment (vmstan) [inline spec/controllers/api/v1/accounts_controller_spec.rb:132]:
```suggestion
          expect(json[:error]).to eq 'Following your own account is not allowed'
```

**Review by vmstan — CHANGES_REQUESTED** [review]

**Comment by mjankowski** [comment]
Closing due to lack of activity, need for rebase, unresponded-to review comments, etc.

If you want to push forward, please rebase, reopen, address comments, etc.

**Pull request closed without merging**